fault-tolerance/src/main/java/io/helidon/faulttolerance/ErrorChecker.java

```diff
@@ -22,25 +22,21 @@
 interface ErrorChecker {
     boolean shouldSkip(Throwable throwable);
 
+    /**
+     * Returns ErrorChecker that skips if throwable is in skipOnSet or if applyOnSet
+     * is not empty and throwable is not in it. Note that if applyOnSet is empty, then
+     * it is equivalent to it containing {@code Throwable.class}.
+     *
+     * @param skipOnSet set of throwables to skip logic on.
+     * @param applyOnSet set of throwables to apply logic on.
+     * @return An error checker.
+     */
     static ErrorChecker create(Set<Class<? extends Throwable>> skipOnSet, Set<Class<? extends Throwable>> applyOnSet) {
-        Set<Class<? extends Throwable>> skipOn = Set.copyOf(skipOnSet);
-        Set<Class<? extends Throwable>> applyOn = Set.copyOf(applyOnSet);
-
-        if (skipOn.isEmpty()) {
-            if (applyOn.isEmpty()) {
-                return throwable -> false;
-            } else {
-                return throwable -> !applyOn.contains(throwable.getClass());
-            }
-        } else {
-            if (applyOn.isEmpty()) {
-                return throwable -> skipOn.contains(throwable.getClass());
-            } else {
-                throw new IllegalArgumentException("You have defined both skip and apply set of exception classes. "
-                                                           + "This cannot be correctly handled; skipOn: " + skipOn
-                                                           + " applyOn: " + applyOn);
-            }
+        return throwable -> containsThrowable(skipOnSet, throwable)
```

Comment:
There is a reason for copying the sets - this method is called from a builder, which has mutable properties. This should work only on the data at the point the method was called and should ignore future mutation of these sets.

Reply:
Make sense, will fix